client/app/nasha/add/nasha-add.service.js

```diff
@@ -0,0 +1,62 @@
+class NashaAddService {
+    constructor ($q, $translate, OrderHelperService, OvhApiOrder, ServiceHelper) {
+        this.$q = $q;
+        this.$translate = $translate;
+        this.OrderHelperService = OrderHelperService;
+        this.OvhApiOrder = OvhApiOrder;
+        this.ServiceHelper = ServiceHelper;
+    }
+
+    getAvailableRegions () {
+        return this.OvhApiOrder.Lexi().schema()
+            .$promise
+            .then(response => _.filter(response.models["dedicated.NasHAZoneEnum"].enum, datacenter => datacenter !== "gra"))
+            .catch(this.ServiceHelper.errorHandler("nasha_order_loading_error"));
+    }
+
+    getOffers () {
+        return this.OvhApiOrder.Cart().Product().Lexi().get({ cartId: "6f111f20-61f3-476c-b38e-58684c7ded7b", productName: "nasha" })
```

Comment:
Oops. Friday code 🤣  !

Reply:
Nah it's fine it's a first draft I just wanted to get CR started